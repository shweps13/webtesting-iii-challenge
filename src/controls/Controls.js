import React from 'react';

const Controls = props => {
  const { locked, closed, toggleLocked, toggleClosed } = props;

  return (
    <div className="controls panel" data-testid="controls-element">
      <button disabled={!closed} onClick={toggleLocked} className="toggle-btn" data-testid="control-btn1">
        {locked ? 'Unlock Gate' : 'Lock Gate'}
      </button>
      <button disabled={locked} onClick={toggleClosed} className="toggle-btn" data-testid="control-btn2">
        {closed ? 'Open Gate' : 'Close Gate'}
      </button>
    </div>
  );
};

export default Controls;
