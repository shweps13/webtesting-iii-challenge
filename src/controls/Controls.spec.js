// Test away
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";

import Controls from './Controls';


test('Dashboard renders without crashing', () => {
  render(<Controls />);
});

test('Matches snapshot', () => {
    const tree = renderer.create(<Controls />); 
    expect(tree.toJSON()).toMatchSnapshot();
});

test('Controls element provide buttons to toggle the closed and locked states', () => {

    const { getByTestId } = render(<Controls />);
    getByTestId(/control-btn1/i);
    getByTestId(/control-btn2/i);
});

test('Buttons text changes to reflect the state the door will be in if clicked', () => {

    const { getByTestId, queryByText } = render(<Controls />);
    const lock = queryByText(/Lock Gate/i);
    fireEvent.click(getByTestId(/control-btn2/i));
    expect(queryByText(/Unlock Gate/i)).toBeTruthy;

});