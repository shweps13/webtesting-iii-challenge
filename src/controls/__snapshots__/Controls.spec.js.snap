// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="controls panel"
  data-testid="controls-element"
>
  <button
    className="toggle-btn"
    data-testid="control-btn1"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="control-btn2"
  >
    Close Gate
  </button>
</div>
`;
